packages/eas-cli/src/update/queries.ts

```diff
@@ -0,0 +1,95 @@
+import chalk from 'chalk';
+import CliTable from 'cli-table3';
+
+import { PaginatedQueryOptions } from '../commandUtils/pagination';
+import { BranchUpdateObject, UpdateQuery } from '../graphql/queries/UpdateQuery';
+import Log from '../log';
+import formatFields from '../utils/formatFields';
+import { printJsonOnlyOutput } from '../utils/json';
+import { paginatedQueryWithConfirmPromptAsync } from '../utils/queries';
+import { UPDATE_COLUMNS, formatUpdate, getUpdateGroupsWithPlatforms } from './utils';
+
+export const UPDATES_LIMIT = 50;
+
+export async function listAndRenderUpdatesOnBranchByNameAsync(
+  projectId: string,
+  branchName: string,
+  options: PaginatedQueryOptions
+): Promise<void> {
+  if (options.nonInteractive || options.limit) {
```

Comment:
```suggestion
  if (options.nonInteractive || options.limit === undefined) {
```

Reply:
Otherwise, `limit=0` is falsy. This is a great example why you should always be explicit :) And yes, I know that `limit` cannot be `0` because of https://github.com/expo/eas-cli/pull/1213/files#diff-31e70cc48acc14099da35dd4184e58264124cd41f89a32b2fbc6b76d99a1e3d5R41
However, this line of code does not guarantee the other line won't change.

Reply:
😛 this code flow is supposed to execute if `options.limit` is truthy, so it is working as intended, and adding the undefined check would flip the meaning of the check. Although now that you've pointed this out, I think we can just remove the limit check altogether. 

When I started working on this a while ago I thought of `limit` as the max items to fetch (total). But it really is the max items to fetch at a time or in other words, the page size. So looking at the code now, there isn't really a reason to drop into the non-interactive flow if it is present. 

Will vet the rest of the PR once again before I merge. Thank you for reviewing